File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        curr=prices[0]
        profit=0

        for i in prices:
            if(i<curr):
                curr=i
            else:
                sell=i-curr
                profit=max(profit,sell)       

        return profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int profit=0;
        int curr=prices[0];

        for(int i=1;i<n;i++){
            if(prices[i]<curr){
                curr=prices[i];
            }
            else{
                int sell=prices[i]-curr;
                profit=Math.max(profit,sell);
            }
        }

        return profit;
    }
}